### ISSUE TITLE
The harbor database should fail early when upgrading from an unsupported version

### ISSUE THREAD

**Issue by heww** [goharbor/harbor#21099]
It will be successful when the customer upgrades the harbor from 2.7 to 2.11, but the customer can't see any previous artifacts from the new harbor instance.

 The harbor database should fail early when upgrading from an unsupported version. 

Here is the /data tree info after the upgrade.

[harbor-data-tree.txt](https://github.com/user-attachments/files/17537829/harbor-data-tree.txt)

**Comment by stonezdj** [comment]
For Harbor tile, the metadata.yml has an attribute minimum_version_for_upgrade: it could used to limit too old version upgrade to a new version.

**Comment by heww** [comment]
> For Harbor tile, the metadata.yml has an attribute minimum_version_for_upgrade, it could used to limit too old version upgrade to a new version.

Yes, agree to limit on the tile side. But harbor-db itself should fail early for this case too. I don't know whether the harbor helm is affected by this issue too.
